### PR TITLE
Fix 482

### LINKED ISSUE #482: Filtering looks broken
Our usage of butterworth looks like it is not doing what is should.

### PR DESCRIPTION
**Pull request by AlexandreAbraham**
Fix #482 by clipping frequency. Include test.

### CODE REVIEW

File: nilearn/signal.py [hunk 1]
```diff
@@ -148,6 +149,17 @@ def _detrend(signals, inplace=False, type="linear", n_batches=10):
     return signals
 
 
+def _check_wn(btype, freq, nyq):
+    wn = freq / nyq
```
Comment (GaelVaroquaux) [inline nilearn/signal.py]: I think that you should cast nyq to a float: float(nyq), just in case some integer finds its way here.

**Comment by GaelVaroquaux** [comment]
Looks great. Thanks heaps. Maybe a better name for wn, lf, hf?

File: nilearn/signal.py (hunk 1 above)
Comment (AlexandreAbraham) [inline nilearn/signal.py]: I can do it but it is already implictly casted in line 216: `nyq = sampling_rate * 0.5`

**Comment by AlexandreAbraham** [comment]
> Maybe a better name for wn, lf, hf?

Reworded.

File: nilearn/signal.py (hunk 1 above)
Comment (GaelVaroquaux) [inline nilearn/signal.py]: > I can do it but it is already implictly casted in line 216: nyq =
> sampling_rate * 0.5

That's outside the function. If someone later reuses the function, he/she
may not be careful enough.

**Comment by AlexandreAbraham** [comment]
@GaelVaroquaux is that okay? Sorry to bother you but Mehdi wants to use filtering and he would like to have this guard ;)

**Comment by GaelVaroquaux** [comment]
Merged. Thanks

**Comment by AlexandreAbraham** [comment]
Cool! Thanks.

**Pull request merged**